scene/main/node.cpp

```diff
@@ -1417,12 +1417,12 @@ TypedArray<Node> Node::find_children(const String &p_pattern, const String &p_ty
 	int ccount = data.children.size();
 	for (int i = 0; i < ccount; i++) {
 		if (p_owned && !cptr[i]->data.owner) {
-			continue;
+			goto next_loop;
 		}
```

Comment:
Regarding this change I think the previous behavior might be intended, see https://github.com/godotengine/godot/pull/75459#issuecomment-1511832318.